scripts/benchmark/checks/deploy_dos.py

```diff
@@ -0,0 +1,296 @@
+import logging
+import pathlib
+from typing import (
+    Tuple
+)
+
+from web3 import (
+    Web3
+)
+
+from eth_utils import (
+    encode_hex,
+    decode_hex,
+)
+
+from eth.constants import (
+    CREATE_CONTRACT_ADDRESS
+)
+from eth.chains.base import (
+    MiningChain,
+)
+from eth.rlp.blocks import (
+    BaseBlock,
+)
+
+from .base_benchmark import (
+    BaseBenchmark,
+)
+from utils.chain_plumbing import (
+    FUNDED_ADDRESS,
+    FUNDED_ADDRESS_PRIVATE_KEY,
+    get_all_chains,
+)
+from utils.compile import (
+    get_compiled_contract
+)
+from utils.reporting import (
+    DefaultStat,
+)
+from utils.tx import (
+    new_transaction,
+)
+
+from abc import (
+    abstractmethod
+)
+
+FIRST_TX_GAS_LIMIT = 367724
+SECOND_TX_GAS_LIMIT = 62050
+THIRD_TX_GAS_LIMIT = 104789
+FORTH_TX_GAS_LIMIT = 21272
+FIFTH_TX_GAS_LIMIT = 21272
+
+CONTRACT_FILE = 'scripts/benchmark/contract_data/DOSContract.sol'
+CONTRACT_NAME = 'DOSContract'
+
+W3_TX_DEFAULTS = {'gas': 0, 'gasPrice': 0}
+
+
+class BaseDOSContractBenchmark(BaseBenchmark):
```

Comment:
I guess that we could even turn some of this into a `BaseContractInteractionBenchmark` class that is reused for the ERC20 token benchmarks as well.

Doesn't need to be addressed in this PR though.